image/image.go

```diff
@@ -112,6 +112,39 @@ func acquireSnap(sto Store, snapName string, dlOpts *downloadOptions) (downloade
 	return downloadSnapWithSideInfo(sto, snapName, dlOpts)
 }
 
+func fetchSnapAssertions(fn string, f *asserts.Fetcher) error {
+	// fetch the snap assertions too
+	sha3_384, _, err := asserts.SnapFileSHA3_384(fn)
+	if err != nil {
+		return err
+	}
+	ref := &asserts.Ref{
+		Type:       asserts.SnapRevisionType,
+		PrimaryKey: []string{sha3_384},
+	}
+	if err := f.Fetch(ref); err != nil {
+		return fmt.Errorf("cannot fetch assertion %q: %s", ref, err)
+	}
+	return nil
+}
+
+func makeFetcher(sto Store, db *asserts.Database) *asserts.Fetcher {
```

Comment:
this can make a local:

type addingFetcher struct {
    *Fetcher
    addedRefs []*Ref
} 

with a AddedRefs() extra method on it